### DISCUSSION TITLE
Translate keys in markdown files

### DISCUSSION THREAD

**Discussion by popuguytheparrot** [facebook/docusaurus#7354] [Q&A]
Hello. Tell me, is it possible to somehow use mdx files as templates without copying files and use keys from translations to json in them?

**Comment by Josh-Cena** [comment]
No-MDX files are always translated as a whole.

You _can_ use `Translate` components in MDX files, like

```mdx
import Translate from "@docusaurus/Translate";

This is some text with <Translate>a translated fragement</Translate>
```

And it will be behave correctly, but you can't use `write-translations` to extract these translatable segments.

**Reply by popuguytheparrot** [reply]
Unpleasant. Are there any solutions in the plans to use the templating approach?

**Reply by Josh-Cena** [reply]
Could you give an example of what you mean by "templating"

**Reply by popuguytheparrot** [reply]
Translations will be in dir i18n/[locale]/docs/guide/start.json 

```
website # Root directory of your site
└── docs
    └── guide
        └── start.md
```
```mdx
---
id: start
title: <Translate>"{#locale-id}" + "start.title"</Translate>
---

## <Translate>"{#locale-id}" + "start.heading"</Translate>

<Translate>"{#locale-id}" + "start.description"</Translate>

```

**Reply by slorber** [reply]
it is not possible to do so atm. MDX React components do not work in frontmatter, and title extraction/TOC generation would likely not work well with `<Translate>` either


At best you can do:

```mdx
---
id: start
title: title
---

## Heading

import MyTranslatedBlock from "@site/src/components/MyTranslatedBlock"

<MyTranslatedBlock/>
```

And you can use `<Translate>` in the MyTranslatedBlock component
